src/components/AllProductsListSection/AllProductsListSection.tsx

```diff
@@ -0,0 +1,95 @@
+'use client';
+
+import GeneralProductCard from '../GeneralProductCard';
+import { useEffect, useState, useCallback } from 'react';
+import { useApi } from '~/hooks/useApi';
+import { useAuthLayout } from '~/hooks/useAuthLayout';
+import { Product } from '~/types/product';
+import lessThan from '~/../public/authenticated/lessThan.svg';
+import greaterThan from '~/../public/authenticated/greaterThan.svg';
+import Image from 'next/image';
+
+const maxProductsPerpage = 6;
+
+export default function AllProductsListSection() {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [maxPage, setMaxPage] = useState<number>(0);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [noProducts, setNoProducts] = useState<boolean>(false);
```

Comment:
I would rename this to something more descriptive like `isProductsEmpty` or `noProductsFound`